IEEE_754/ieee.js

```diff
@@ -0,0 +1,34 @@
+//IEEE-754 convertor
+
+var bit_32 = 0xFFFFFFFF;
+var bit_23 = 0x7FFFFF;
+var bit_8  = 0xFF;
+
+function print(){
+    var ieee = document.getElementById("hexieee").value;
+    if (ieee[1] != "x"){
+        ieee = "0x" + ieee;
+    }
+    ieee &= bit_32;
+
+    var sign = (ieee >> 31) & 0x1;
+    var exponent = ((ieee >> 23) & bit_8) - 0x7F;
+    var significand = ieee & bit_23;
+    var sigval = 1;
+    var output;
+    if (exponent == -127 || exponent == 128){
+        document.getElementById("warning").innerHTML = "WARNING - Invalid Exponent value" + "<br />" + "Typically all 0 or 1 values for exponents are reserved for numbers that can't be " + "<br />" + "represented by IEEE 754 such as 0, NaM, infinity or denorm";
```

Comment:
It should matter here, however, in general you should avoid using [innerHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML) as it is a potential cross-site scripting vulnerability. You should add container elements to the page, style them with properly, and then use textContent to change their content.